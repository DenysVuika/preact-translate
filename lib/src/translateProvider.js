import { createContext } from 'preact';

import useTranslate from './useTranslate';

const TranslateContext = createContext();

const TranslateProvider = props => {
  const { t, setLang, lang, isReady } = useTranslate({
    root: props.root
  });

  return (
    <TranslateContext.Provider
      value={{
        t,
        setLang,
        lang,
        isReady
      }}
    >
      {props.children}
    </TranslateContext.Provider>
  );
};

export { TranslateProvider, TranslateContext };
