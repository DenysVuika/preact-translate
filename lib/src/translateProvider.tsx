import { Context, createContext, h } from 'preact';
import { LanguageData } from './languageData';
import { TranslateOptions } from './translateOptions';
import useTranslate from './useTranslate';

const TranslateContext = createContext(null) as Context<
  ReturnType<typeof useTranslate>
>;

const defaultOptions: TranslateOptions = {
  root: 'assets',
  lang: 'en'
};

export interface TranslateProviderProps {
  root?: string;
  lang?: string;
  fallbackLang?: string;
  translations?: LanguageData;
  children?: any;
}

const TranslateProvider = (props: TranslateProviderProps) => {
  const { t, setLang, lang, isReady } = useTranslate(
    {
      root: props.root || defaultOptions.root,
      lang: props.lang || defaultOptions.lang,
      fallbackLang: props.fallbackLang || defaultOptions.fallbackLang
    },
    props.translations
  );

  return (
    <TranslateContext.Provider
      value={{
        t,
        setLang,
        lang,
        isReady
      }}
    >
      {props.children}
    </TranslateContext.Provider>
  );
};

export { TranslateProvider, TranslateContext };
