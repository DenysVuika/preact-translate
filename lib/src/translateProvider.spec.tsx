import { h } from 'preact';
import { render } from 'preact-testing-library';
import { TranslateProvider } from './translateProvider';

describe('TranslateProvider', () => {
  it('matches snapshot', () => {
    expect(
      render(
        <TranslateProvider>
          <div />
        </TranslateProvider>
      )
    ).toMatchSnapshot();
  });
});
