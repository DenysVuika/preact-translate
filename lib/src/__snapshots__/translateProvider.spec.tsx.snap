// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslateProvider matches snapshot 1`] = `
Object {
  "container": <div>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
